src/tools/rbd/action/Import.cc

```diff
@@ -458,9 +485,15 @@ int do_import_diff(librados::Rados &rados, librbd::Image &image,
       std::cerr << "rbd: error opening " << path << std::endl;
       return r;
     }
+
+    r = get_import_file_size(path, fd, size);
+    if (r < 0) {
+      goto done;
+    }
   }
-  r = do_import_diff_fd(rados, image, fd, no_progress, 1, sparse_size);
+  r = do_import_diff_fd(rados, image, fd, no_progress, 1, sparse_size, size);
```

Comment:
I'm not sure I understand the need to bother with the size of the physical disk for `rbd import-diff`.  IIUC this size is used only for progress reporting and the block device can be much larger than the import-diff "file" on it?  If correct, there is no need to introduce `get_import_file_size`.

Reply:
It's used by ``do_import_v2`` and I was initially aiming to cover the v2 format as well. Since that's currently unsupported, I guess we could drop this.

Reply:
Yes, please.  And let's ensure a clear error message for the v2 case (same as for export).